app/models/training_period.rb

```diff
@@ -46,6 +46,7 @@ class TrainingPeriod < ApplicationRecord
   # Delegations
   delegate :name, to: :delivery_partner, prefix: true, allow_nil: true
   delegate :name, to: :lead_provider, prefix: true, allow_nil: true
+  delegate :name, to: :expression_of_interest_lead_provider, prefix: :expression_of_interest_lead_provider, allow_nil: true
```

Comment:
Would it be more straightforward to just call `training_period.expression_of_interest_lead_provider.name` in the component, than to add another delegation?